posts/map_blocks_example.md

```diff
@@ -0,0 +1,179 @@
+    """
+    Compute MLD based on ∆σ criterion. Uses xarray.map_blocks.
+    
+    Parameters
+    ----------
+    
+    SALT : xarray.DataArray
+      Salinity
+    TEMP : xarray.DataArray
+      Potential temperature
+    dsigma : float, optional
+      The value for ∆σ.
+      
+    Returns
+    -------
+    
+    MLD : xarray.DataArray
+      The MLD (m) defined as the point in the water column where
+      density exceeds rho[0] + dsigma.      
+    """
+    
+    # determine dimensionality
+    dims_in = SALT.dims
+    assert dims_in == TEMP.dims, 'dimension mismatch'
+    assert 'z_t' in SALT.coords, 'z_t not found in SALT coords'
+
+    # drop ancillary coordinates (this may not be necessary)
+    SALT = SALT.reset_coords(drop=True)
+    TEMP = TEMP.reset_coords(drop=True)
+    
+    # compute density
+    rho = pop_tools.eos(SALT.chunk({'z_t': 10}), 
+                        TEMP.chunk({'z_t': 10}), 
+                        depth=SALT.z_t * 0.).compute()
+    
+    # add these coordinates which creep in somewhere, I think when xarray does the unstack
+    # without these, I get an error: not expecting {'nlat', 'nlon'}
+    # chunking here is arbitrary and maybe suitable for the global POP_gx1v7 grid
+    if 'nlat' in rho.dims and 'nlon' in rho.dims:
+        rho = rho.assign_coords({
+            'nlat': xr.DataArray(np.arange(len(SALT.nlat)), dims=('nlat')),
+            'nlon': xr.DataArray(np.arange(len(SALT.nlon)), dims=('nlon')),
+        })
+        
+    rho = rho.chunk(rho_chunks).persist()
+    
+    # compute and return MLD - this is where the parallelization comes in
+    return xr.map_blocks(
+        interp_mld_dsigma, rho,
+        kwargs=dict(dsigma=dsigma), 
+        template=rho.isel(z_t=0).drop('z_t'),
+    )
+```
+
+### Applying Mixed Layer Depth Calculation
+This function, `interp_mld_dsigma` is designed to be called within `xr.map_blocks`. It assumes that `rho_in` has a depth dimension, `z_t` and some number of unspecified other dimensions, `non_vertical_dims`.
+
+The non_vertical_dims are "stacked" using `xarray.stack` and the code loops over these dimensions, performing linear interpolation in `z_t` at each location.
+
+The data are return unstacked with the dimensions ordered as the arrived in `rho_in`.
+
+We define the function here, then immediately call it to test it; arbitrarily, we're calling it on TEMP rather than density.
+
+```python
+def interp_mld_dsigma(rho_in, dsigma=0.03):
+    """compute MLD at point using interpolation"""
+    
+    # TODO: this will probably fail if non_vertical_dims = []
+    #       In that case, we might use `expand_dims` to add a dummy dimension
+    non_vertical_dims = [d for d in rho_in.dims if d not in ['z_t']] 
+    rho_stack = rho_in.stack(non_vertical_dims=non_vertical_dims)
+    mld_stack = xr.full_like(rho_stack.isel(z_t=slice(0, 1)), fill_value=np.nan)
+    N = len(rho_stack.non_vertical_dims)
+    z_t = rho_in.z_t * 1e-2
```

Comment:
remove the `* 1e-2`

This converts from `cm` to `m`, but above we set the units to `SALT.z_t.attrs['units']`. I think it's more straightforward to leave the units as they are.